Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1273,6 +1273,52 @@ public extension ChatChannelController {
             }
         }
     }
+    
+    /// Loads the given number of pinned messages based on pagination parameter in the current channel.
+    ///
+    /// - Parameters:
+    ///   - pageSize: The # of pinned messages to load. Equals to `25` by default.
+    ///   - sorting: The sorting options. By default, results are sorted descending by `pinned_at` field.
+    ///   - pagination: The pagination parameter. If `nil` is provided, most recently pinned messages are fetched.
+    ///   - completion: The completion to be called on **callbackQueue** when request is completed.
+    func loadPinnedMessages(
+        pageSize: Int = .messagesPageSize,
+        sorting: [Sorting<PinnedMessagesSortingKey>] = [],
+        pagination: PinnedMessagesPagination? = nil,
+        completion: @escaping (Result<[ChatMessage], Error>) -> Void
+    ) {
+        guard let cid = cid, isChannelAlreadyCreated else {
+            channelModificationFailed { completion(.failure($0 ?? ClientError.ChannelNotCreatedYet())) }
+            return
+        }
+        
+        let query = PinnedMessagesQuery(
+            pageSize: pageSize,
+            sorting: sorting,
+            pagination: pagination
+        )
+        
+        updater.loadPinnedMessages(in: cid, query: query) {
+            switch $0 {
+            case let .success(payload):
+                var messages: [ChatMessage] = []
+                
+                self.client.databaseContainer.write({ session in
+                    messages = payload
+                        .messages
+                        .map { .init(payload: $0, session: session) }
```

Comment:
Why don't we save the messages to the DB first, and then map to the model like we usually do? 🤔

Reply:
The same reason why message search results should not be persisted: when saved to the database they are linked to the channel which leads to gaps in the local channel history and breaks the pagination (if we have `0..<20` messages locally and get `100th` pinned message, we will show `0..<20, 99` in channel and the next page we will be loading is `100..<120`)

Reply:
Isn't storing them in the DB an issue then? Are we allowing gaps with this implementation?

Reply:
We don't save messages to the database. The session is passed to model initializer so we can populate fields (isSentByCurrentUser) that are not included into the payload.

Reply:
I think we already have something to fix that issue:
```swift
// Some pinned messages might be in the local database, but should not be fetched
// if they do not belong to the regular channel query.
let ignoreOlderMessagesPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [
    .init(format: "channel.oldestMessageAt == nil"),
    .init(format: "createdAt >= channel.oldestMessageAt")
])
```

Reply:
I remember I fixed this issue long time ago 🙂 It's not the cleanest solution, but it works

Reply:
okay, I was not aware we have this 🤔 It means we're safe to save messages from search/pinned ones to database actually 🙂 cc @tbarbugli

Reply:
@nuno-vieira is right, we can do this later on and keep the current solution to avoid gaps